### ISSUE TITLE
Support `git diftool -d`, AKA Git directory diff

### ISSUE THREAD

**Issue by segevfiner** [microsoft/vscode#174790]
When VS Code is set as the difftool for Git, make it work nicely with `git difftool -d`, AKA a directory diff. VS Code doesn't know how to handle that scenario well ATM.

**Comment by saierd** [comment]
I think now that the multi-diff editor exists, this feature could potentially be implemented by treating a directory given to `--diff` like a list of files for the multi-diff editor.

This would be straightforward for files which exist in both directories. For a complete implementation of a directory diff one would obviously need to handle moved files as well, which is trickier. Maybe ignoring moves and just showing a diff to an empty file in that case would be enough for a start.

From looking around the code a bit (I am not familiar with the code base), it seems like this feature could go somewhere [around here](https://github.com/microsoft/vscode/blob/9d4274e5599f6ba2c667a54ec0d1d6594d30145b/src/vs/workbench/electron-sandbox/window.ts#L989-L991). When `pathsToEditors` discovers that the paths are directories, read the list of files in those directories and give them to `openResources`, which can then create a `IResourceMultiDiffEditorInput` for opening the multi-diff editor.

Would such an implementation be acceptable as a contribution?

**Comment by tyleo** [comment]
I agree with @dpward's opinion that closing all of the existing issues is overzealous, though I could see some being closed as duplicates.

`git difftool --dir-diff {lhs} {rhs}` is the most common way I use `git difftool` these days. I've eagerly awaited this functionality in VSCode. It is a delightful feature in [Araxis Merge](https://www.araxis.com/merge/index.en) and currently the only reason I prefer it over VSCode for diff.

I also agree with @saierd's recommendation to support this with the [multi diff editor](https://code.visualstudio.com/updates/v1_86#_review-multiple-files-in-diff-editor). It looks like a great solution and would hit the bar for replacing competing tools in my workflow.

@saierd, I'm curious if you tried implementing your suggestion. While it may not be a goal or golden path for the VSCode team, it looks like the normal course of iteration may have sleepwalked close to a viable solution to this request. I'm hopeful that you've discovered a path forward that folks can agree on. I'd be new to VSCode development but willing to provide development bandwidth here if there is anything I can do to help move this along.
